src/main/java/uk/gov/ons/census/casesvc/utility/RedactHelper.java

```diff
@@ -0,0 +1,57 @@
+package uk.gov.ons.census.casesvc.utility;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Modifier;
+import java.util.Arrays;
+
+public class RedactHelper {
+  private static final String REDACTION_FAILURE = "Failed to redact sensitive data";
+  private static final String REDACTION_TARGET = "setUac";
+  private static final String REDACT_TO_THIS_VALUE = "REDACTED";
+  private static final ObjectMapper objectMapper = ObjectMapperFactory.objectMapper();
+
+  public static Object redact(Object rootObjectToRedact) {
+    try {
+      Object rootObjectToRedactDeepCopy =
+          objectMapper.readValue(
+              objectMapper.writeValueAsString(rootObjectToRedact), rootObjectToRedact.getClass());
+      recursivelyRedact(
+          rootObjectToRedactDeepCopy,
+          REDACTION_TARGET,
+          rootObjectToRedactDeepCopy.getClass().getPackageName());
+      return rootObjectToRedactDeepCopy;
+    } catch (JsonProcessingException e) {
+      throw new RuntimeException(REDACTION_FAILURE, e);
+    }
+  }
+
+  private static void recursivelyRedact(Object object, String methodTofind, String packageName) {
+    Arrays.stream(object.getClass().getMethods())
+        .filter(item -> Modifier.isPublic(item.getModifiers()))
+        .forEach(
+            method -> {
+              if (method.getName().startsWith("get")
+                  && method.getReturnType().getPackageName().equals(packageName)) {
+                try {
+                  Object invokeResult = method.invoke(object);
+                  if (invokeResult != null) {
+                    recursivelyRedact(invokeResult, methodTofind, packageName);
```

Comment:
Is there a danger of infinite recursion if two classes reference each other?

Reply:
yes it will do that, but that would be caught in unit or integration testing. I haven't tried to write a helper class for use in any and all Comp Sci scenarios, but instead a redactor for use on this project